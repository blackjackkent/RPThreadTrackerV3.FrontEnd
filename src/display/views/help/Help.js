import React from 'react';
import { Row, Col } from 'reactstrap';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import AboutTrackerPane from './components/AboutTrackerPane';
import SupportGuidesPane from './components/SupportGuidesPane';
import ContactPane from './components/ContactPane';
import tabs from '../../../infrastructure/constants/tabs';
import TabContent from '../../shared/styled/TabContent';
import StaticTabNav from '../../shared/static/StaticTabNav';
import StaticDropdownNav from '../../shared/static/StaticDropdownNav';
import FAQPane from './components/FAQPane';

const propTypes = {
	match: PropTypes.shape({
		url: PropTypes.string,
		params: PropTypes.shape({
			tabId: PropTypes.string
		})
	}).isRequired
};

function mapStateToProps() {
	return {};
}

const Help = (props) => {
	const { match } = props;
	const options = Object.values(tabs.HELP);
	return (
		<div className="animated fadeIn static-container help-container">
			<Row>
				<Col className="d-lg-none text-center">
					<StaticDropdownNav
						data-spec="help-static-dropdown-nav"
						activeTab={match.url}
						options={options}
					/>
				</Col>
			</Row>
			<Row>
				<Col className="d-none d-lg-block" md={3}>
					<StaticTabNav data-spec="help-static-tab-nav" options={options} />
				</Col>
				<Col xs="12" lg="9">
					<TabContent activeTab={match.params.tabId}>
						<AboutTrackerPane />
						<SupportGuidesPane />
						<FAQPane />
						<ContactPane />
					</TabContent>
				</Col>
			</Row>
		</div>
	);
};

Help.propTypes = propTypes;
export default connect(mapStateToProps)(Help);
