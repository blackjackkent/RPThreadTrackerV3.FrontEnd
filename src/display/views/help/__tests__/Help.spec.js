// #region imports
import React from 'react';
import { getSpecWrapper, shallowWithState } from '~/testhelpers/helpers.unit';
import Help from '../Help';
// #endregion imports

// #region mocks
jest.mock('../components/AboutTrackerPane', () => 'AboutTrackerPane');
jest.mock('../components/SupportGuidesPane', () => 'SupportGuidesPane');
jest.mock('../components/ContactPane', () => 'ContactPane');
jest.mock('../components/FAQPane', () => 'FAQPane');
jest.mock('../../../../infrastructure/actions', () => ({}));
jest.mock('../../../../infrastructure/constants/tabs', () => ({
	HELP: {
		TAB1: {
			href: '/help/tab1',
			name: 'Tab 1'
		},
		TAB2: {
			href: '/help/tab2',
			name: 'Tab 2'
		}
	}
}));
jest.mock('../../../shared/static/StaticTabNav', () => 'StaticTabNav');
jest.mock('../../../shared/static/StaticDropdownNav', () => 'StaticDropdownNav');
// #endregion mocks

const createTestProps = (propOverrides) => ({
	match: {
		url: '/help/tab1',
		params: {
			tabId: 'tab1'
		}
	},
	...propOverrides
});

const createTestState = (stateOverrides) => ({
	...stateOverrides
});

describe('rendering', () => {
	describe('snapshots', () => {
		it('should render valid snapshot', () => {
			const props = createTestProps();
			const state = createTestState();
			const jsx = <Help {...props} />;
			const element = shallowWithState(jsx, state).dive();
			expect(element).toMatchSnapshot();
		});
	});
	describe('options', () => {
		it('should populate options from imported tabs on dropdown nav', () => {
			const props = createTestProps();
			const state = createTestState();
			const jsx = <Help {...props} />;
			const element = shallowWithState(jsx, state).dive();
			const form = getSpecWrapper(element, 'help-static-dropdown-nav');
			const { options } = form.props();
			expect(options).toHaveLength(2);
		});
		it('should populate options from imported tabs on tab nav', () => {
			const props = createTestProps();
			const state = createTestState();
			const jsx = <Help {...props} />;
			const element = shallowWithState(jsx, state).dive();
			const form = getSpecWrapper(element, 'help-static-tab-nav');
			const { options } = form.props();
			expect(options).toHaveLength(2);
		});
	});
});
