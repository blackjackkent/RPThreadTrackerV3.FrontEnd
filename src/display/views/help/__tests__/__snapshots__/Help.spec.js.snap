// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering snapshots should render valid snapshot 1`] = `
<div
  className="animated fadeIn static-container help-container"
>
  <Row
    tag="div"
  >
    <Col
      className="d-lg-none text-center"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      <StaticDropdownNav
        activeTab="/help/tab1"
        data-spec="help-static-dropdown-nav"
        options={
          Array [
            Object {
              "href": "/help/tab1",
              "name": "Tab 1",
            },
            Object {
              "href": "/help/tab2",
              "name": "Tab 2",
            },
          ]
        }
      />
    </Col>
  </Row>
  <Row
    tag="div"
  >
    <Col
      className="d-none d-lg-block"
      md={3}
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      <StaticTabNav
        data-spec="help-static-tab-nav"
        options={
          Array [
            Object {
              "href": "/help/tab1",
              "name": "Tab 1",
            },
            Object {
              "href": "/help/tab2",
              "name": "Tab 2",
            },
          ]
        }
      />
    </Col>
    <Col
      lg="9"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
      xs="12"
    >
      <Styled(TabContent)
        activeTab="tab1"
      >
        <AboutTrackerPane />
        <SupportGuidesPane />
        <FAQPane />
        <ContactPane />
      </Styled(TabContent)>
    </Col>
  </Row>
</div>
`;
